video2dataset/dataloader/video_decode.py

```diff
@@ -98,10 +98,9 @@ def get_frames(self, reader, n_frames, stride, **kwargs):  # pylint: disable=arg
         if n_frames * stride > len(reader):
             if not self.pad_frames:
                 raise ValueError("video clip not long enough for decoding")
-            n_frames = len(reader) // stride
-
+            n_frames = len(reader)
         # sample frame start and choose scene
-        if n_frames == len(reader):
+        if n_frames == len(reader) or n_frames == len(reader) // stride:
```

Comment:
why are you adding these changes?

Reply:
if you set `n_frames=None` and fps to like 2, above in another part of the decoder it sets `n_frames = len(reader) // stride`, which makes this if statement not trigger and it randomly chooses a start frame, which I don't want. so I added this extra `or` to take into account cases where we subsample fps.